bug: Early exit from function with out arg that uses a channel causes seg fault

This is a memory management bug that has been tracked into our callDestructors
pass.  If an out argument was assigned to before exiting the function then
destructor calls for the channel and file were always inserted before the
function returned, even though in the early exit case the channel and file in
question were (to the user's perspective) not even in existence.  Thus, when
they were destroyed, the program would segfault.

To ensure that this program always fails while the problem is present (the
segfault appearance for each individual portion only occurs under either --devel
or --no-devel mode but not both, so it would be possible for one to be fixed but
not the other), test it under valgrind, where the unnecessary free will always
be visible.

See also outArgEarlyExit[ Reader | Writer ] for individual odd behavior.

Tom's AMM branch does not exhibit this behavior, so when that gets merged into
trunk, this future should be resolved.
